### PR TITLE
feat: Add /connectedNodes endpoint

### PR DESCRIPTION
**Pull request by andrewelamb**
fixes [fds-824](https://sagebionetworks.jira.com/browse/FDS-824)

### CODE REVIEW

File: apps/schematic/api/schematic_api/openapi/openapi.yaml [hunk 1]
```diff
@@ -823,6 +871,39 @@ components:
...
       - '@lombok.Builder'
+    RelationshipType:
+      description: A type of schema relationship
+      example: requiresDependency
+      type: string
+    Relationship:
+      description: A schema relationship.
+      properties:
+        component1:
+          description: The dsiaplay name of the first component in the relationship.
```
Comment (mialy-defelice) [inline apps/schematic/api/schematic_api/openapi/openapi.yaml:882]: typo

File: apps/schematic/api/schematic_api/openapi/openapi.yaml [hunk 2]
```diff
@@ -823,6 +871,39 @@ components:
...
+          example: Component1
+          title: component1
+          type: string
+        component2:
+          description: The diaplay name of the second component in the relationship.
```
Comment (mialy-defelice) [inline apps/schematic/api/schematic_api/openapi/openapi.yaml:887]: typo

File: libs/schematic/api-description/src/components/schemas/Relationship.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+type: object
+description: A schema relationship.
+properties:
+  component1:
+    type: string
+    description: The dsiaplay name of the first component in the relationship.
```
Comment (mialy-defelice) [inline libs/schematic/api-description/src/components/schemas/Relationship.yaml:6]: Typo

File: libs/schematic/api-description/src/components/schemas/Relationship.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    example: Component1
+  component2:
+    type: string
+    description: The diaplay name of the second component in the relationship.
```
Comment (mialy-defelice) [inline libs/schematic/api-description/src/components/schemas/Relationship.yaml:10]: typo

File: apps/schematic/api/schematic_api/controllers/schema_controller.py [hunk 5]
```diff
@@ -48,6 +49,21 @@ def get_property_label(
...
+    Gets a subgraph containing all relationships of a given type # noqa: E501
+
+    :param relationship_type: Type of relationship in a schema, such as requiresDependency
+    :type relationship_type: str
+    :param schema_url: The URL of a schema in jsonld form
+    :type schema_url: str
+
+    :rtype: Union[RelationshipsPage, Tuple[RelationshipsPage, int], Tuple[RelationshipsPage, int, Dict[str, str]]
+    """
+    return schema_controller_impl.get_relationships(relationship_type, schema_url)
```
Comment (mialy-defelice) [inline apps/schematic/api/schematic_api/controllers/schema_controller.py:64]: To me, get relationships implies something a bit different than what is happening here. 
Here you are retrieving the nodes that are connected by a given relationship type.

File: apps/schematic/api/schematic_api/openapi/openapi.yaml [hunk 6]
```diff
@@ -823,6 +871,39 @@ components:
...
       x-java-class-annotations:
       - '@lombok.Builder'
+    RelationshipType:
+      description: A type of schema relationship
+      example: requiresDependency
+      type: string
+    Relationship:
+      description: A schema relationship.
+      properties:
+        component1:
```
Comment (mialy-defelice) [inline apps/schematic/api/schematic_api/openapi/openapi.yaml:881]: These are not necessarily components, node would be a better descriptor.

**Review by mialy-defelice — CHANGES_REQUESTED** [review]

File: apps/schematic/api/schematic_api/controllers/schema_controller.py (hunk 5 above)
Comment (andrewelamb) [reply apps/schematic/api/schematic_api/controllers/schema_controller.py:64]: The endpoint retrieves a pair of nodes connected by a relationship right? What would you call the two nodes then?

Comment (mialy-defelice) [reply apps/schematic/api/schematic_api/controllers/schema_controller.py:64]: lol, I left out a suggestion bc I couldn't think of anything elegant... but maybe something like, get_connected_nodes_by_relationship? are you limited by characters?

Comment (mialy-defelice) [reply apps/schematic/api/schematic_api/controllers/schema_controller.py:64]: to me a endpoint called get_relationships, I would expect to retrieve the types of relationships between node pairs,...which we can actually do, and why I got confused earlier. I thought you had created a second endpoint where you were supplying two nodes and were retrieving the relationship, but then I realized you were just specifying the returns.

Comment (andrewelamb) [reply apps/schematic/api/schematic_api/controllers/schema_controller.py:64]: I think ConnectedNodes would be fine (RelationshipType would be a parameter)

**Comment by andrewelamb** [comment]
@mialy-defelice Could you check the changes I made?

File: apps/schematic/api/schematic_api/controllers/schema_controller_impl.py [hunk 7]
```diff
@@ -10,9 +10,73 @@
...
+from schematic_api.models.connected_nodes_page import ConnectedNodesPage
+from schematic_api.models.connected_nodes import ConnectedNodes
 from schematic_api.controllers.utils import handle_exceptions
 
 
+def get_connected_nodes_from_schematic(
+    relationship_type: str,
+    schema_url: str,
+) -> list[list[Any]]:
+    """Gets the relationships asked for
```
Comment (mialy-defelice) [inline apps/schematic/api/schematic_api/controllers/schema_controller_impl.py:22]: I would update this to something along the lines of "gets nodes connected via provide relationship"

File: apps/schematic/api/schematic_api/controllers/schema_controller.py [hunk 8]
```diff
@@ -5,13 +5,29 @@
...
+from schematic_api.models.connected_nodes_page import ConnectedNodesPage  # noqa: E501
 from schematic_api.models.node_properties_page import NodePropertiesPage  # noqa: E501
 from schematic_api.models.nodes_page import NodesPage  # noqa: E501
 from schematic_api.models.validation_rules_page import ValidationRulesPage  # noqa: E501
 from schematic_api import util
 from schematic_api.controllers import schema_controller_impl
 
 
+def get_connected_nodes(schema_url, relationship_type):  # noqa: E501
+    """Gets a subgraph containing all relationships of a given type
```
Comment (mialy-defelice) [inline apps/schematic/api/schematic_api/controllers/schema_controller.py:17]: sorry I didn't catch this earlier, but for this endpoint, it does call function that returns a subgraph, but the endpoint itself it returning a list(?) of node pairs.

**Review by mialy-defelice — CHANGES_REQUESTED** [review]
Just a couple small changes

**Comment by andrewelamb** [comment]
@mialy-defelice Could you check again? :)

**Review by mialy-defelice — APPROVED** [review]

**Pull request merged**